src/functionalTest/java/uk/gov/hmcts/divorce/solicitor/SolicitorSubmitPetitionTest.java

```diff
@@ -57,4 +69,52 @@ public void shouldUpdateCaseDataWithOrderSummaryAndAddSolCaseRolesWhenIssueFeeIs
             STRICT
         );
     }
+
+    @Test
+    public void shouldChangeStateToAwaitingPaymentConfirmationIfValidCaseData()
+        throws Exception {
+        Response response = RestAssured
+            .given()
+            .relaxedHTTPSValidation()
+            .baseUri(testUrl)
+            .header(CONTENT_TYPE, APPLICATION_JSON_VALUE)
+            .header(SERVICE_AUTHORIZATION, generateServiceAuthTokenFor(s2sName))
+            .header(HttpHeaders.AUTHORIZATION, generateIdamTokenForSolicitor())
+            .body(
+                CallbackRequest
+                    .builder()
+                    .eventId(SOLICITOR_STATEMENT_OF_TRUTH_PAY_SUBMIT)
+                    .caseDetails(
+                        CaseDetails
+                            .builder()
+                            .id(createCaseInCcd().getId())
+                            .data(caseDataWithStatementOfTruth())
+                            .state(SOTAgreementPayAndSubmitRequired.getName())
+                            .build()
+                    )
+                    .build()
+            )
+            .when()
+            .post(ABOUT_TO_SUBMIT_CALLBACK_URL);
+
+        assertThat(response.getStatusCode()).isEqualTo(OK.value());
+
+        assertEquals(
+            expectedCcdCallbackResponse("classpath:responses/ccd-callback-solicitor-submit-petition-about-to-submit.json"),
+            response.asString(),
+            STRICT
+        );
+    }
+
+    private Map<String, Object> caseDataWithStatementOfTruth() {
+        Map<String, Object> caseData = new HashMap<>();
```

Comment:
I wonder if it would be easier to maintain if we loaded this from a json file?

Reply:
Yes, I think it would be.  Added commit to do this.